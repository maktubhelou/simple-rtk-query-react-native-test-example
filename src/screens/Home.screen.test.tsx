import 'react-native';
import React from 'react';
import { renderWithProviders } from '../../test-utils';
import { Home } from './Home.screen';
import { server } from '../test/server';
import { rest } from 'msw';

describe('Home', () => {
  it('renders "Bulbasaur"', async () => {
    const component = <Home />;
    const { findByText } = renderWithProviders(component);

    await findByText('Loading...');

    const bulbasaur = await findByText('Bulbasaur');
    expect(bulbasaur).toBeTruthy();
  });

  it('handles error response', async () => {
    server.use(
      rest.get(
        'https://pokeapi.co/api/v2/pokemon/bulbasaur',
        (req, res, ctx) => {
          return res(ctx.status(500));
        },
      ),
    );

    const component = <Home />;
    const { findByText } = renderWithProviders(component);
    const errorText = await findByText('Oh no, there was an error!');

    expect(errorText).toBeTruthy();
  });
});
