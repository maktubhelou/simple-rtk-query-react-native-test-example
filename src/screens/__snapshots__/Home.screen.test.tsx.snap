// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Pokemon
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                Object {
                  "color": "#444",
                },
              ]
            }
          >
            Select your favorite 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              Pokemon 
            </Text>
            from the list below.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                Object {
                  "color": "#000",
                },
              ]
            }
          >
            Example
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                Object {
                  "color": "#444",
                },
              ]
            }
          >
            <View>
              <Text>
                Bulbasaur
              </Text>
              <Image
                source={
                  Object {
                    "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "width": 100,
                  }
                }
              />
            </View>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
